src/contract/certificateStore.js

```diff
@@ -0,0 +1,119 @@
+}
+
+CertificateStore.prototype.name = function _name() {
+  return this.contract.methods.name().call();
+};
+
+CertificateStore.prototype.verificationUrl = function _verificationUrl() {
+  return this.contract.methods.verificationUrl().call();
+};
+
+CertificateStore.prototype.owner = function _owner() {
+  return this.contract.methods.owner().call();
+};
+
+CertificateStore.prototype.getIssuedBlock = function _getIssuedBlock(
+  merkleRoot
+) {
+  return this.contract.methods.getIssuedBlock(merkleRoot).call();
+};
+
+CertificateStore.prototype.isBatchIssued = function _isBatchIssued(merkleRoot) {
+  return this.contract.methods.isBatchIssued(merkleRoot).call();
+};
+
+CertificateStore.prototype.isRevoked = function _isRevoked(merkleRoot) {
+  return this.contract.methods.isRevoked(merkleRoot).call();
+};
+
+CertificateStore.prototype.transferOwnership = function _transferOwnership(
+  newOwner
+) {
+  return this.contract.methods
+    .transferOwnership(newOwner)
+    .send({ from: this.account })
+    .then(tx => {
+      const { status, transactionHash } = tx;
+      if (web3.utils.hexToNumber(status) === 1) {
+        return tx;
+      } else if (web3.utils.hexToNumber(status) === 0) {
+        return Promise.reject(
+          new Error(`Transaction has failed, check tx hash: ${transactionHash}`)
+        );
+      }
+      return Promise.reject(new Error(`Unknown transaction status: ${tx}`));
+    });
+};
+
+CertificateStore.prototype.issueBatch = function _issueBatch(hash) {
+  return this.contract.methods
+    .issueBatch(hash)
+    .send({ from: this.account })
+    .then(tx => {
+      const { status, transactionHash } = tx;
+      if (web3.utils.hexToNumber(status) === 1) {
+        return tx;
+      } else if (web3.utils.hexToNumber(status) === 0) {
+        return Promise.reject(
+          new Error(`Transaction has failed, check tx hash: ${transactionHash}`)
+        );
+      }
+      return Promise.reject(new Error(`Unknown transaction status: ${tx}`));
+    });
+};
+
+CertificateStore.prototype.revokeClaim = function _revokeClaim(
+  merkleRoot,
+  claim,
+  reason
+) {
+  return this.contract.methods
+    .revokeClaim(merkleRoot, claim, reason)
+    .send({ from: this.account })
+    .then(tx => {
+      const { status, transactionHash } = tx;
+      if (web3.utils.hexToNumber(status) === 1) {
+        return tx;
+      } else if (web3.utils.hexToNumber(status) === 0) {
+        return Promise.reject(
+          new Error(`Transaction has failed, check tx hash: ${transactionHash}`)
+        );
+      }
+      return Promise.reject(new Error(`Unknown transaction status: ${tx}`));
+    });
+};
+
+CertificateStore.prototype.deployStore = function _deployStore(
+  verificationUrl,
+  name
+) {
+  return this.contract
+    .deploy({
+      data: bytecode,
+      arguments: [verificationUrl, name]
+    })
+    .send({ from: this.account, gas: this.gas, gasPrice: this.gasPrice })
```

Comment:
Does web3 have any utility to estimate gas needed? Otherwise we would have to manually configure the gas? Ditto for Gas price.